### PR TITLE
Added destructuing guide

### PR DESCRIPTION
**Pull request by mzavarella30**

### CODE REVIEW

**Comment by puredanger** [comment]
Probably easiest to view those comments at:
https://github.com/mzavarella30/clojure-site/commit/16057b932ec6b7700a1877164767d1294ef4e4b4

If you want to give it another pass on those, I'd be happy to merge it.

**Comment by puredanger** [comment]
ping?

**Comment by puredanger** [comment]
@mzavarella30 where are we on this? is it ready for another look or still wip?

**Comment by mzavarella30** [comment]
Working on the final section now! I'll submit a pull request later tonight

File: content/guides/destructuring.adoc [hunk 1]
```diff
@@ -0,0 +1,534 @@
...
+  (println "You can reach him at" phone)
+  (println "He lives at" street city state zip)
+  (println "Maybe you can write to him about" fav-hobby "or" second-hobby))
+
+(print-contact-info john-smith)
+;= John Smith is the Sith Lord of Git at Functional Industries
+;= You can reach him at 2150041776
+;= He lives at 452 Lisp Ln. Macroville Kentucky 81321
+;= Maybe you can write to him about running or hiking
+----
```
Comment (mzavarella30) [inline content/guides/destructuring.adoc]: Alright, here we go! I tried to make the code examples a little more realistic and in depth. I also added a little bit of sequential destructuring to the final example for the sake of having an all-inclusive final showing. Let me know what you think!

**Comment by puredanger** [comment]
I will work on getting this integrated along with a number of changes - thanks!

**Comment by puredanger** [comment]
I made a lot of edits, but the destructuring guide is now live - thanks!

**Pull request merged**